src/memray/_memray/tracking_api.h

```diff
@@ -213,10 +213,22 @@ class Tracker
         }
         RecursionGuard guard;
 
-        NativeTrace trace;
+        std::optional<NativeTrace> trace{std::nullopt};
         if (Tracker::areNativeTracesEnabled()) {
+            auto t_trace_data_ptr = static_cast<std::vector<NativeTrace::ip_t>*>(
+                    pthread_getspecific(s_native_unwind_vector_key));
+            if (!t_trace_data_ptr) {
+                t_trace_data_ptr = new std::vector<NativeTrace::ip_t>();
+                if (pthread_setspecific(s_native_unwind_vector_key, t_trace_data_ptr) != 0) {
+                    fprintf(stderr, "memray: pthread_setspecific failed\n");
```

Comment:
Should we deactivate tracking here?

Reply:
Sure, I suppose. I think there's very few ways this could happen:
1. We've got a bug that causes us to fail to create the key before we use it
2. We (or the program we're tracking) has a memory corruption bug that trashes our key
3. We've gotten an ENOMEM trying to allocate TLS for a new thread

I think any of those are pretty fantastically unlikely, but I suppose none of them are really recoverable, so deactivating tracking is fine. (Technically we could recover from (1) by falling back to a new vector for each call, but it doesn't seem worth the effort, given that it would represent some bug in our code.)